src/Polykey.ts

```diff
@@ -418,7 +399,7 @@ export default class Polykey {
           try {
             const vaultKey = Buffer.from(this.metadata.vaults[vaultName].key)
             const vault = new Vault(vaultName, vaultKey, this.polykeyPath)
-            this.vaults.set(vaultName, vault)
+            this.vaultStore.setVault(vaultName, vault)
```

Comment:
I can see that `Polykey` is one of the primary classes in this library. However its domain modelling is too broad. `Polykey` as a class shouldn't be dealing with secrets directly. That should be in a subdomain of  vaults.

Instead `Polykey` should be managing vaults, and capabilities, and peers. Each of which has their own subdomain. We can discuss this more later.

Reply:
Solved in PR 40.